types/express-serve-static-core/index.d.ts

```diff
@@ -562,11 +562,12 @@ export interface Request<
     secure: boolean;
 
     /**
-     * Return the remote address, or when
-     * "trust proxy" is `true` return
-     * the upstream addr.
+     * Return the remote address.
+     *
+     * If "trust proxy" is provided and none of the upstream addresses
+     * meet the conditions, then `undefined` will be returned.
```

Comment:
When you say, "...and none of the upstream addresses meet the conditions", to what conditions are you referring? I think it'll be beneficial if we make that more explicit 🙂 

We should also add that the value may be `undefined` if the socket is destroyed.

Reply:
> There's nothing in the Express.js [documentation](https://expressjs.com/en/4x/api.html#req.ip) (or the link provided) that explicitly mentions this property can be undefined.

Yes, I was wrong about the real reason why the ip might be undefined. I will remove this line and add comment about `req.socket.remoteAddress`.